pkg/network/config/sysctl/sysctl_darwin.go

```diff
@@ -0,0 +1,69 @@
+// +build darwin
+
+package sysctl
+
+import (
+	"encoding/binary"
+
+	"strings"
+	"syscall"
+	"time"
+)
+
+type sctl struct {
+	ttl      time.Duration
+	lastRead time.Time
+	path     string
+}
+
+func newSCtl(sysctl string, cacheFor time.Duration) *sctl {
+	return &sctl{
+		ttl:  cacheFor,
+		path: sysctl,
+	}
+}
+
+func (s *sctl) get(now time.Time) (string, bool, error) {
+	if !s.lastRead.IsZero() && s.lastRead.Add(s.ttl).After(now) {
+		return "", false, nil
+	}
+
+	content, err := syscall.Sysctl(s.path)
+	if err != nil {
+		return "", false, err
+	}
+
+	s.lastRead = now
+	return strings.TrimSpace(content), true, nil
+}
+
+// Int16 represents a 16 bit int sysctl
+type Int16 struct {
+	*sctl
+	v uint16
+}
```

Comment:
nit: type definitions should probably be at the top of the file

Reply:
in the linux file, each type was defined above the related methods, so that the individual types were together with the methods.  I kept that pattern.